nutanix/data_source_nutanix_pbr.go

```diff
@@ -0,0 +1,428 @@
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"address_type": {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+											},
+										},
+									},
+									"source": {
+										Type:     schema.TypeList,
+										Computed: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"address_type": {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+											},
+										},
+									},
+									"priority": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+									"protocol_parameters": {
+										Type:     schema.TypeList,
+										Computed: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"tcp": {
+													Type:     schema.TypeList,
+													Computed: true,
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"destination_port_range_list": portRangeSchema(),
+															"source_port_range_list":      portRangeSchema(),
+														},
+													},
+												},
+												"udp": {
+													Type:     schema.TypeList,
+													Computed: true,
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"destination_port_range_list": portRangeSchema(),
+															"source_port_range_list":      portRangeSchema(),
+														},
+													},
+												},
+												"icmp": {
+													Type:     schema.TypeList,
+													Computed: true,
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"icmp_type": {
+																Type:     schema.TypeInt,
+																Computed: true,
+															},
+															"icmp_code": {
+																Type:     schema.TypeInt,
+																Computed: true,
+															},
+														},
+													},
+												},
+												"protocol_number": {
+													Type:     schema.TypeInt,
+													Computed: true,
+												},
+											},
+										},
+									},
+									"action": {
+										Type:     schema.TypeList,
+										Computed: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"action": {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+											},
+										},
+									},
+									"protocol_type": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceNutanixPbrRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*Client).API
+
+	pbrUUID, ok := d.GetOk("pbr_uuid")
+	if !ok {
+		return diag.Errorf("please provide pbr reference uuid")
+	}
+
```

Comment:
Doesnt api support query with name as well ? Or any other API we can use incase to query using name

Reply:
Can we use filter query from list api ?

Reply:
no , api doesn't support query with name. I have introduced filter incase if any other attribute supports.